src/main/java/com/flowingcode/vaadin/addons/gridexporter/BaseInputStreamFactory.java

```diff
@@ -60,14 +63,36 @@ protected Stream<T> getDataStream(Query newQuery) {
     }
     return stream;
   }
-  
-  protected List<Pair<String,Column<T>>> getGridHeaders(Grid<T> grid) {
-    return exporter.columns.stream().map(column -> ImmutablePair.of(GridHelper.getHeader(grid,column),column))
+
+  protected List<Pair<String, Column<T>>> getGridHeaders(Grid<T> grid) {
+    return exporter.columns.stream().map(column -> buildPair(grid, column))
         .collect(Collectors.toList());
   }
 
-  protected List<Pair<String,Column<T>>> getGridFooters(Grid<T> grid) {
-    return exporter.columns.stream().map(column -> ImmutablePair.of(GridHelper.getFooter(grid,column),column))
+  private Pair<String, Column<T>> buildPair(Grid<T> grid, Column<T> column) {
+    String header = (String) ComponentUtil.getData(column, GridExporter.COLUMN_HEADER);
+    if (Strings.isBlank(header)) {
```

Comment:
Shouldn't it be `header!=null`? (so that I can configure `""` as a custom column header for export)